src/artemis/tests/test_fgs_communicator.py

```diff
@@ -0,0 +1,265 @@
+
+DUMMY_TIME_STRING = "1970-01-01 00:00:00"
+GOOD_ISPYB_RUN_STATUS = "DataCollection Successful"
+BAD_ISPYB_RUN_STATUS = "DataCollection Unsuccessful"
+
+test_start_document = {
+    "uid": "d8bee3ee-f614-4e7a-a516-25d6b9e87ef3",
+    "time": 1666604299.6149616,
+    "versions": {"ophyd": "1.6.4.post76+g0895f9f", "bluesky": "1.8.3"},
+    "scan_id": 1,
+    "plan_type": "generator",
+    "plan_name": "run_gridscan_and_move",
+}
+test_event_document = {
+    "descriptor": "bd45c2e5-2b85-4280-95d7-a9a15800a78b",
+    "name": "ispyb_motor_positions",
+    "time": 1666604299.828203,
+    "data": {
+        "slit_gaps_xgap": 0.1234,
+        "slit_gaps_ygap": 0.2345,
+        "synchrotron_machine_status_synchrotron_mode": "test",
+        "undulator_gap": 1.234,
+    },
+    "timestamps": {"det1": 1666604299.8220396, "det2": 1666604299.8235943},
+    "seq_num": 1,
+    "uid": "29033ecf-e052-43dd-98af-c7cdd62e8173",
+    "filled": {},
+}
+test_stop_document = {
+    "run_start": "d8bee3ee-f614-4e7a-a516-25d6b9e87ef3",
+    "time": 1666604300.0310638,
+    "uid": "65b2bde5-5740-42d7-9047-e860e06fbe15",
+    "exit_status": "success",
+    "reason": "",
+    "num_events": {"fake_ispyb_params": 1, "primary": 1},
+}
+test_failed_stop_document = {
+    "run_start": "d8bee3ee-f614-4e7a-a516-25d6b9e87ef3",
+    "time": 1666604300.0310638,
+    "uid": "65b2bde5-5740-42d7-9047-e860e06fbe15",
+    "exit_status": "fail",
+    "reason": "",
+    "num_events": {"fake_ispyb_params": 1, "primary": 1},
+}
+test_descriptor_document = {
+    "uid": "bd45c2e5-2b85-4280-95d7-a9a15800a78b",
+    "run_start": "d8bee3ee-f614-4e7a-a516-25d6b9e87ef3",
+}
+
+
+def test_fgs_communicator_reset():
+    communicator = fgs_communicator.FGSCommunicator()
+    assert communicator.processing_time == 0.0
+    communicator.params.detector_params.prefix = "file_name"
+    assert communicator.params == FullParameters()
+
+    communicator.results = "some position to move to"
+    communicator.reset(FullParameters())
+    assert communicator.results is None
+
+
+@patch("artemis.fgs_communicator.run_start")
+@patch("artemis.fgs_communicator.run_end")
+@patch("artemis.fgs_communicator.wait_for_result")
+@patch("artemis.fgs_communicator.StoreInIspyb3D.store_grid_scan")
+@patch("artemis.fgs_communicator.StoreInIspyb3D.get_current_time_string")
+@patch(
+    "artemis.fgs_communicator.StoreInIspyb3D.update_grid_scan_with_end_time_and_status"
+)
+def test_run_gridscan_zocalo_calls(
+    mock_ispyb_update_time_and_status: MagicMock,
+    mock_ispyb_get_time: MagicMock,
+    mock_ispyb_store_grid_scan: MagicMock,
+    wait_for_result: MagicMock,
+    run_end,
+    run_start,
+):
+
+    dc_ids = [1, 2]
+    dcg_id = 4
+
+    mock_ispyb_store_grid_scan.return_value = [dc_ids, None, dcg_id]
+    mock_ispyb_get_time.return_value = DUMMY_TIME_STRING
+    mock_ispyb_update_time_and_status.return_value = None
+
+    params = FullParameters()
+    params.detector_params.prefix += str(time.time())
```

Comment:
Nit: Why is this required? We should probably just make sure the test isn't writing nexus files by mocking things out

Reply:
Okay, I just figured I should mock the minimum possible to be able to run the test, I guess I like the idea that it is calling the real nexus writer and not a fake one. But now that you mention it, since it's not actually what we're testing here I will remove it